src/pudl/analysis/record_linkage/eia_ferc1_record_linkage.py

```diff
@@ -35,27 +35,264 @@
+                    col1="installation_year_ferc1",
+                    col2="installation_year_eia",
+                    output_name="installation_year",
+                )
+            ],
+            columns=["installation_year_ferc1", "installation_year_eia"],
+        ),
+    }
+
+
+@op
+def get_compiled_input_manager(plants_all_ferc1, fbp_ferc1, plant_parts_eia):
+    """Get :class:`InputManager` object with compiled inputs for model."""
+    inputs = InputManager(plants_all_ferc1, fbp_ferc1, plant_parts_eia)
+    # compile/cache inputs upfront. Hopefully we can catch any errors in inputs early.
+    inputs.execute()
+    return inputs
+
+
+@op(out={"all_pairs_df": Out(), "train_pairs_df": Out()})
+def get_pairs_dfs(inputs):
+    """Get a dataframe with all possible FERC to EIA record pairs.
+
+    Merge the FERC and EIA records on ``block_col`` to generate possible
+    record pairs for the matching model. First do this with all record
+    pairs, then with only the training data records.
+
+    Arguments:
+        inputs: :class:`InputManager` object.
+
+    Returns:
+        all_pairs_df: Dataframe with all possible record pairs
+        train_pairs_df: Dataframe with all possible record pairs from the
+            training data.
+    """
+    ferc1_df = inputs.get_plants_ferc1().reset_index()
+    eia_df = inputs.get_plant_parts_eia_true().reset_index()
+    block_col = "plant_id_report_year_util_id"
+    all_pairs_df = ferc1_df.merge(
+        eia_df, how="inner", on=block_col, suffixes=("_ferc1", "_eia")
+    ).set_index(["record_id_ferc1", "record_id_eia"])
+    ferc1_train_df = inputs.get_train_ferc1().reset_index()
+    eia_train_df = inputs.get_train_eia().reset_index()
+    block_col = "plant_id_report_year_util_id"
+    train_pairs_df = ferc1_train_df.merge(
+        eia_train_df, how="inner", on=block_col, suffixes=("_ferc1", "_eia")
+    ).set_index(["record_id_ferc1", "record_id_eia"])
+    return (all_pairs_df, train_pairs_df)
+
+
+@op
+def get_y_label_df(train_pairs_df, inputs):
+    """Get the dataframe of y labels.
+
+    For each record pair in ``train_pairs_df``, a 0 if the pair is not
+    a match and a 1 if the pair is a match.
+    """
+    label_df = np.where(
+        train_pairs_df.merge(
+            inputs.get_train_df(),
+            how="left",
+            left_index=True,
+            right_index=True,
+            indicator=True,
+        )["_merge"]
+        == "both",
+        1,
+        0,
+    )
+    return label_df
+
+
+@op
+def get_best_matches_with_overwrites(match_df, inputs):
+    """Get dataframe with the best EIA match for each FERC record."""
+    return find_best_matches(match_df).pipe(overwrite_bad_predictions, inputs.train_df)
+
+
+@op
+def run_matching_model(features_train, features_all, y_df):
+    """Run model to match EIA to FERC records."""
+    return run_model(
+        features_train=features_train,
+        features_all=features_all,
+        y_df=y_df,
+    )
+
+
+@op
```

Comment:
@katie-lamb I see what you mean, I tried using the `graph_multi_asset` and specifying the io-manager through the `outs`, but it's still not actually writing to the DB for some reason. I did you find you can specify the io-manager for the final `op` in the graph, and that works. IDK if this is actually better than just using the second asset, but it's an option.

```suggestion
@op(out={"out_pudl__yearly_assn_eia_ferc1_plant_parts": Out(io_manager_key="pudl_sqlite_manager_key"})
```

I tested this locally and I was able to remove the wrapper asset and get this to write to the DB as expected. Either way is a bit janky, it would be nice if dagster just let us specify the asset manager directly in a `graph_asset`.

Reply:
Ya I think I like this better, just because then there's no confusion with what out_pudl__yearly... vs _out_pudl__yearly... is. Will make this change.